src/System.Net.Primitives/src/System/Net/NetworkCredential.cs

```diff
@@ -162,9 +149,9 @@ internal bool IsEqualTo(object compObject)
                 return false;
             }
 
-            return (InternalGetUserName() == compCred.InternalGetUserName() &&
-                    InternalGetDomain() == compCred.InternalGetDomain() &&
-                    string.Equals(_password, compCred._password, StringComparison.Ordinal));
+            return (_userName == compCred.UserName &&
+                    _domain == compCred.Domain &&
+                    string.Equals(_password, compCred.Password, StringComparison.Ordinal));
```

Comment:
This is incorrect if a change like the one I suggested earlier isn't done, as it'll potentially be comparing the wrong field.  That said, this whole function is DEBUG-only and doesn't appear to have any callers, so maybe the whole function can just be deleted.